pkgs/misc/vim-plugins/overrides.nix

```
@@ -711,6 +711,35 @@ self: super: {
     dependencies = with super; [ bclose-vim ];
   });
 
+  nvim-typescript = let
+     # node2nix dynamically resolves nodeDependencies
+     nvim-typescript-deps = (callPackage (
+       runCommand "nvim-typescript-rplugin.nix" {
+         buildInputs = [ nodePackages.node2nix ];
+       } ''
+         mkdir -p $out
+         cd ${super.nvim-typescript.src}/rplugin/node/nvim_typescript
+         node2nix --input package.json \
+                  --lock package-lock.json \
+                  --include-peer-dependencies \
+                  --nodejs-10 \
+                  --development \
+                  --output $out/node-packages.nix \
+                  --node-env $out/node-env.nix \
+                  --composition $out/default.nix
+       ''
+     ) {});
```

this should fix your evaluation error
```suggestion
     nvim-typescript-deps = runCommand "nvim-typescript-rplugin.nix" {
         buildInputs = [ nodePackages.node2nix ];
       } ''
         mkdir -p $out
         cd ${super.nvim-typescript.src}/rplugin/node/nvim_typescript
         node2nix --input package.json \
                  --lock package-lock.json \
                  --include-peer-dependencies \
                  --nodejs-10 \
                  --development \
                  --output $out/node-packages.nix \
                  --node-env $out/node-env.nix \
                  --composition $out/default.nix
       '';
```

---

actually, now that I look at this again, I'm not sure this is correct either. I'm not familiar with using node within nix to give you affirmative answer.

---

No, I think this is not going to work without committing a generated node2nix file.